File: frontend/src/types/customer.ts
export interface CustomerRead {
  id: string;
  name: string;
  email: string;
  phone: string | null;
  billing_address: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface CustomerCreate {
  name: string;
  email: string;
  phone?: string | null;
  billing_address: string;
}

export interface CustomerUpdate {
  name?: string;
  email?: string;
  phone?: string | null;
  billing_address?: string;
}

File: backend/app/schemas/customer.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CustomerCreate(BaseModel):
    name: str
    email: str
    phone: str | None = None
    billing_address: str


class CustomerUpdate(BaseModel):
    name: str | None = None
    email: str | None = None
    phone: str | None = None
    billing_address: str | None = None


class CustomerRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    email: str
    phone: str | None
    billing_address: str
    is_active: bool
    created_at: datetime
    updated_at: datetime
